src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java

```diff
@@ -174,22 +173,21 @@ public NBTTagCompound serialize(@NotNull final IFactoryController controller, @N
         public StandardRequests.DeliveryRequest getNewInstance(
                                                                 @NotNull final Delivery input,
                                                                 @NotNull final IRequester location,
-                                                                @NotNull final IToken token,
+                                                                @NotNull final IToken<?> token,
                                                                 @NotNull final RequestState initialState)
         {
             return new StandardRequests.DeliveryRequest(location, token, initialState, input);
         }
     }
 
-    @SuppressWarnings(Suppression.BIG_CLASS)
     public static final class ToolRequestFactory implements IRequestFactory<Tool, StandardRequests.ToolRequest>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 41 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2972)